contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,231 @@
+
+```typescript
+@Component({
+  selector: 'app-person',
+  templateUrl: './person.component.html',
+  styleUrls: ['./person.component.css']
+})
+export class PersonComponent {
+  person: Person = {
+    name: 'John Doe'
+  };
+  constructor() { }
+}
+```
+
+The `@Component` tag is used to mark a component. It provides the metadata that determines how the component is processed during runtime.
+
+* The use of `selector: app-person` is how Angular identifies this component in a template. This is an example of a custom tag used in Angular templates. The following code constitutes the template of `app.component.html`.
+
+    ```html
+    <h1>Person Data</h1>
+    <app-person></app-person>
+    <hr/>
+    ```
+
+    Angular identifies the `<app-person>` tag, instantiates the `PersonComponent` and renders the view defined by the template `./person.component.html`.
+
+* The `person.name` used in the template `./person.component.html`, refers to the name property defined in the above component.
+
+### Dependency Injection
+
+Now let us look at what **Dependency Injection**(DI) means. Dependencies are services that a class needs in order to perform its function. We can visualize this with the help of an example.
+
+<center>
+<img src="dependency.png" width="60%">
+
+_Figure 1. What is a dependency?_ <sup>[source](https://medium.freecodecamp.org/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f)</sup>
+</center>
+
+Dependency injection is a way of providing a class with the required services. The official website claims that the DI framework increases the "efficiency and modularity" of an app. Given below is a an injectable service class.
+ 
+```typescript
+import { Injectable } from '@angular/core';
+import { PERSONS } from './mock-persons';
+@Injectable({
+  providedIn: 'root',
+})
+export class PersonService {
+  getPersons() { return PERSONS; }
+}
+```
+
+The `PersonComponent` declared previously, can make use of this `PersonService` by injecting the dependency in its constructor as a parameter.
+
+```
+constructor(personService: PersonService) {
+ this.persons = personService.getPersons();
+}
+```
+
+In the above example, we write the `getPersons` method only once in the service class and we can access this method from any component by injecting the service in the constructor.
+This is how DI increases "efficiency and modularity".
+
+Declarative Templates and Dependency Injection are only a couple of noteworthy features of Angular. A more comprehensive [list](https://angular.io/features) is available on the official website. 
+ 
+## Why Angular?
+
+Now that we know what Angular is, let us look at some benefits it has to offer.
+
+### Benefit 1: Angular Provides Good Tooling Services.
```

Comment:
Not sure if the full stops at the end of the heading are necessary. Looks odd to me, but your call :)